### DISCUSSION TITLE
Images don't show in AutoCAD until you manually Reload them in the xref manager

### DISCUSSION THREAD

**Discussion by worlds6440** [mozman/ezdxf#845] [FAQ]
Images with *relative paths* fail to show on DXFs opened in AutoCAD by default. XRef manager shows them as *!unreadable*. You have to go to the xref manager and manually 'Reload' them before they show.

##POTENTIAL FIX:
I found that you need to add the following to the headers section

'''
9
$PROJECTNAME
  1

'''

##REASON:
If the value after the *1* is left blank, it uses the DXF file's path as the initial place to look when loading images and shows the images straight away.

**Comment by mozman** [comment]
setting header vars is easy:

    yourdrawing.header['$PROJECTNAME'] = ''

DXF Reference says to $PROJECTNAME:

> Assigns a project name to the current drawing. Used when an external reference or image is not found on its original path. The project name points to a section in the registry that can contain one or more search paths for each project name defined. Project names and their search directories are created from the Files tab of the Options dialog box

I will add $PROJECTNAME='' as default value in my templates files.

Thanks for your contribution.

Manfred

**Comment by mozman** [comment]
I saw $PROJECTNAME='' is the default in my template files, I will add a HOWTO section for this problem with existing DXF files.

**Comment by mozman** [comment]
Now I got no error message on opening drawings, but the image is still not shown, reference status 'Nicht lesbar' (not readable), I have to run 'Alle Referenzen neu laden' (reload all references) to see the image.

Any suggestion about this? Because the DXF reference does not have ANY information about such internal state problems or how to organize this internal structures correct.

**Comment by worlds6440** [comment]
## Progress
After extensive tests, this appears to be a DXF reading ~~bug~~feature with AutoCAD and images with a relative path or no path.  
  
If the filename in the ```IMAGEDEF``` contains the full path (*absolute* in AutoCAD) then it shows on loading, otherwise it won't display (reports as *unreadable*) until you manually reload using XREF manager.  
  
I performed the same tests when loading identical files (but resaved as DWGs) and they loaded and displayed the images, even with relative / no paths.  
  
## Summary
I haven't found a *fix* for images with a relative path in DXF. However, a workaround appears to be to save the full file path in the DXF or save it as a DWG.

**Comment by mozman** [comment]
Thank you, I will add this information to the howto section.

**Comment by ardamaks** [comment]
Guys, this issue is still persistent in ACAD 2021.
How can be solved. Thanks

**Comment by luzader** [comment]
This is still a problem with DWG TrueView 2023. 

Has anyone else noticed that very short simple image file names seem to avoid this problem? Once I ensured that the image file's name was short and had no special characters (letters, numbers and underscores only) the problem seemed to go away. I didn't rigorously analyze the behavior as its very time consuming.

Also: You can safely put the image in a subdirectory and use a relative path. The name of the subdirectory does not seem to trigger this problem, provided that the image file name itself is very short and simple.

Also pro tip: The XRef manager exists in DWG TrueView, but access to it is only possible if you have a completely broken reference. Create a DXF with a reference to a non-existent file, then the error dialog will let you open the XRef Manager. Once it is open you can pin it and it will be open next time, even if you have no broken references.

**Reply by mozman** [reply]
The XREF manager exist only in DWG TrueView 2023 and maybe later!

**Comment by mozman** [comment]
@luzader Thank you!

If you don't mind, I'll add your tips to the [HowTo](https://ezdxf.mozman.at/docs/howto/viewer.html#show-images-xrefs-on-loading-in-autocad) section.
